### ISSUE TITLE
Make AddPooledDbContextFactory register DbContext in addition to the and factory

### ISSUE THREAD

**Issue by araxemy** [dotnet/efcore#26528]
In #25164 it was suggested that EF Core implicitly registers a scoped `DbContext` as well when `AddDbContextFactory` is used, which I thought was a great idea, and it was done in RC1 I think.

But why not do the same thing for `AddPooledDbContextFactory` too?! I'm using `AddPooledDbContextFactory` in my project and I was surprised to see that the same thing isn't done in that method as well.

**Comment by msallin** [comment]
Same situation here. We use `AddPooledDbContextFactory` in all projects but we also use the EF HealthChecks, which expect the `DbContext` as ctor parameter. Hence, we also have to call  `AddDbContextPool<T>`.

**Comment by aradalvand** [comment]
Hi there, any explanation as to why this was removed from 7.0.0? It seems like it would be relatively straight-forward and simple to implement.

**Comment by ajcvickers** [comment]
@aradalvand We had to punt a number of smaller issues. While each in itself would not take a long time, that time quickly adds up. This one is also only sugar--it's easy to obtain the same functionality by registering trice--which makes it lower priority.

**Comment by xamir82** [comment]
@ajcvickers Hi, could this please be added to the v8 milestone? It doesn't take too much effort to implement but the API is just incomplete without this. Thanks.

**Comment by ErikEJ** [comment]
@xamir82 the project accepts pull requests 😉

**Comment by ajcvickers** [comment]
@xamir82 

> API is just incomplete without it

Can you explain a bit more what you mean by this? To me, this is a bit of sugar that means people who need this don't have to add an extra line of code to register the service they want.

**Comment by schmellow** [comment]
What would be the most correct workaround here?
Would something like that be enough:
```
builder.Services.AddScoped<MyContext>(p =>
{
    return p.GetRequiredService<IDbContextFactory<MyContext>>().CreateDbContext();
});
```
(it does not seem wrong to me from lifecycle perspective at a glance, but maybe i'm missing something)
Or do you have to do full AddDbContextPool round, repeating all the configuration? 
Or maybe the other way around, as was suggested in here https://github.com/npgsql/efcore.pg/issues/3375#issuecomment-2509746639 ?

**Comment by stevendarby** [comment]
@schmellow that is the suggestion here https://learn.microsoft.com/en-us/ef/core/performance/advanced-performance-topics?tabs=with-di%2Cexpression-api-with-constant#managing-state-in-pooled-contexts (see last code snippet in that section)

**Comment by roji** [comment]
AddDbContextFactory already registers both IDbContextFactory and DbContext in DI (#25164), but AddPooledDbContextFactory indeed does not - I've opened #35484 to track that.

In the meantime, it should be fine to simply also call AddDbContext() after the call - without passing configuration to it (it should pick up the configuration already passed to AddPooledDbContextFactory):

```c#
services
    .AddPooledDbContextFactory<BlogContext>(o => o.UseSqlServer(...))
    .AddDbContext<BlogContext>();
```

**Comment by stevendarby** [comment]
@roji why do you need a new issue? I know the title isn't clear but the first post seems quite clear they are asking for this in `AddPooledDbContextFactory`, after seeing it in AddDbContextFactory`. My only concern is things like up-votes get lost when this happens...

**Comment by roji** [comment]
You're right, I looked at the title and didn't notice that the original post was specifically talking AddPooledDbContextFactory, and that AddDbContextFactory was working fine. I'll reverse the situation.

**Comment by susickypavel** [comment]
@roji Based on the labels I believe this was released as a part of .NET 10 preview 1?

Until .NET 10 is released it is ok for lower versions to register the `TContext` with following code?

```
services.AddPooledDbContextFactory<BlogContext>
services.TryAddScoped(sp => sp.GetRequiredService<IDbContextFactory<BlogContext>>().CreateDbContext());
```

Will be the `BlogContext` returned from the pool if request through DI this way? Thank you for clarification.

**Comment by roji** [comment]
@susickypavel yes.

**Comment by JohnYoungers** [comment]
It might be a good idea to call this out as a notable change here: https://learn.microsoft.com/en-us/ef/core/what-is-new/ef-core-10.0/breaking-changes

All of our applications register the `DbContextFactory` by default (even if they may not use it) opposed to `DbContext`: it looks like we had an app that injected a scoped level value into `DbContext`, which had been working fine with the pre-v10 setup, since they never injected `DbContextFactory`; it took a bit to dig into why that was happening.

It wouldn't have been an issue had they originally marked the factory as being at the scope level, but it was a bit confusing as to why that had been working and suddenly stopped
